src/js/core/methods/SignTransaction.js

```diff
@@ -122,22 +128,43 @@ export default class SignTransaction extends AbstractMethod {
         const { device, params } = this;
 
         let refTxs: RefTransaction[] = [];
+        const useLegacySignProcess = device.unavailableCapabilities['replaceTransaction'];
         if (!params.refTxs) {
             // initialize backend
-            const hdInputs = params.inputs.map(inputToHD);
-            const refTxsIds = getReferencedTransactions(hdInputs);
+            const refTxsIds = getReferencedTransactions(params.inputs);
             if (refTxsIds.length > 0) {
                 // validate backend
                 isBackendSupported(params.coinInfo);
                 const blockchain = await initBlockchain(params.coinInfo, this.postMessage);
-                const bjsRefTxs = await blockchain.getReferencedTransactions(refTxsIds);
-                refTxs = transformReferencedTransactions(bjsRefTxs);
+                const rawTxs = await blockchain.getTransactions(refTxsIds);
+                refTxs = transformReferencedTransactions(rawTxs, params.coinInfo);
+
+                const origTxsIds = getOrigTransactions(params.inputs, params.outputs);
+                if (!useLegacySignProcess && origTxsIds.length > 0) {
+                    const rawOrigTxs = await blockchain.getTransactions(origTxsIds);
+                    let addresses = params.addresses;
+                    // sender account addresses not provided
+                    // fetch account info from the blockbook
+                    if (!addresses) {
+                        // TODO: validate inputs address_n's === same account
+                        const node = await device.getCommands().getHDNode(params.inputs[0].address_n.slice(0, 3), params.coinInfo);
+                        const account = await blockchain.getAccountInfo({
+                            descriptor: node.xpubSegwit || node.xpub,
+                            coin: params.coinInfo.name,
+                            details: 'tokens',
```

Comment:
just out of curiosity. is 'tokens' needed here? wouldnt 'basic' suffice?

Reply:
pulling respective part of documentation here: 
 - `basic` (default) return only account balances, without any derived addresses or transaction history
 - `tokens` - response with derived addresses (Bitcoin-like accounts) and ERC20 tokens (Ethereum-like accounts), subject of `tokens` param
  - `tokenBalances` - same as `tokens` with balances, subject of `tokens` param
  - `txs` - `tokenBalances` + complete account transaction history

Reply:
basic will not return derived addresses and addresses is what we want here :)

Reply:
Ah, I got confused by the "tokens" keyword